modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/TransactionStateResolver.java

```diff
@@ -213,18 +215,20 @@ private void resolveTxStateFromTxCoordinator(
         updateLocalTxMapAfterDistributedStateResolved(txId, txMetaFuture);
 
         InternalClusterNode coordinator = coordinatorId == null ? null : clusterNodeResolver.getById(coordinatorId);
-
         if (coordinator == null) {
             // This means the coordinator node have either left the cluster or restarted.
             markAbandoned(txId);
 
             resolveTxStateFromCommitPartition(txId, commitGrpId, txMetaFuture);
         } else {
-            txMessageSender.resolveTxStateFromCoordinator(coordinator.name(), txId, timestamp)
+            txMessageSender.resolveTxStateFromCoordinator(coordinator, txId, timestamp)
                     .whenComplete((response, e) -> {
-                        if (e == null) {
-                            txMetaFuture.complete(asTransactionMeta(response.txStateMeta()));
+                        if (e == null && response.txStateMeta() != null) {
+                            txMetaFuture.complete(asTransactionMeta(Objects.requireNonNull(response.txStateMeta())));
                         } else {
+                            if (e != null && e.getCause() instanceof RecipientLeftException) {
```

Comment:
Are you sure that `RecipientLeftException` is the only exception that is possible here?
As far as I know aforementioned exception will be thrown if there's a node with given name, however it has another id. What about node absence?